tianshou/highlevel/experiment.py

```diff
@@ -491,14 +491,26 @@ def _get_optim_factory(self) -> OptimizerFactory:
         else:
             return self._optim_factory
 
-    def build(self, add_seeding_info_to_name: bool = False) -> Experiment:
-        """Creates the experiment based on the options specified via this builder.
+    def build(
+        self,
+        num_experiments: int = 1,
+        add_seeding_info_to_name: bool = False,
```

Comment:
Default should be True, I think. What happens if it's false and n>1, don't we get name collisions of files?

Reply:
Probably it's better to default to True. If `num_experiments` is >1, it will be set to True internally anyway.

Reply:
Agree :)

Reply:
This violates the principles of the builder pattern and leads to ugly usage patterns. The `build` method must take no arguments. You are conflating the idea of a factory for multiple experiments with the builder. This is a bad idea, which [defies expectations](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

Instead, what you want is 
 * an abstraction (specifically, a factory) which creates an experiment for a given seed (e.g. `SeededExperimentFactory` with `create_experiment(seed)` or sth along those lines). The experiment builder usage would then appear in the respective "create" method.
 * an abstraction which implements a particular way of creating multiple experiments from a base random seed, taking a `SeededExperimentFactory` as an argument. This can handle your current way of creating multiple experiments, which is, ultimately, just one (largely arbitrary) way of creating multiple experiments.

This would be the clean way to do it without abusing design patterns and without contrived mechanisms.
In general: Don't add contrived mechanisms to an existing abstraction; add new abstractions that build upon the existing abstractions.

The argument `add_seeding_info_to_name` is already bad design. The builder has a method `with_name` which should handle the naming. It is rather strange to add a second way of influencing the naming - and the only reason it exists is because the builder pattern was abused.

Reply:
Let's have a call on how to handle this best. The seeding, the new dir structrure, and the multiple experiments should be the new default. I feel like we might get a bloated interface if we have a builder as well as a factory (the latter also needs to be built). I wouldn't want our high-level examples to start with something like 

```python
SeededExperimentFactory.from_experiment_builder(experiment_builder).create_default_seeded_experiments()
```

It's a bit scary and might be off-putting for newcomers

Reply:
Why do you want to change the high-level examples in the first place? The examples are fine the way they are.

What we need is not a change in the existing high-level examples but additional functionality for the case where a user wants to run a series of experiments - and for this case, a new abstraction is required.

Reply:
> the multiple experiments should be the new default

Why? Initially, a user will usually want to run only a single experiment and see what happens. Once the user knows which effects he wants to look into, he may want to conduct a series of experiments - but he might want to vary many things. Assuming that he wants to vary the seeds only is a random assumption to me.

Reply:
In principle, the idea is to be able to evaluate a given configuration on multiple seeds. The examples are, after all, only examples. People who will properly evaluate algorithms on a task will probably build their own structure on their own configuration stack. Most probably, and despite being an anti-pattern, using for example hydra :)

Reply:
What if, instead of adding a function to build a number of experiments, we only provide some functionality that returns the corresponding experiment/sampling configs? Then it's just a matter of either looping over `build` with updated experiment/sampling configs (or newly generated builders), or prividing a launcher with a set of configs.